llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -8340,7 +8344,7 @@ static SDValue extractShiftForRotate(SelectionDAG &DAG, SDValue OppShift,
 // The IsRotate flag should be set when the LHS of both shifts is the same.
 // Otherwise if matching a general funnel shift, it should be clear.
 static bool matchRotateSub(SDValue Pos, SDValue Neg, unsigned EltSize,
```

Comment:
```suggestion
static bool matchRotateSubAdd(SDValue Pos, SDValue Neg, unsigned EltSize,
```

Reply:
I don't think this rename makes sense. The `Sub` here reflects how the shift amounts are related to each-other not something about the combining operation. The fact that we can now handle ADD cases in this function doesn't mean we should have an `Add` suffix on the function, just as the fact that we already handle OR cases doesn't mean we should have an `Or` suffix.